hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILECrck.py

```diff
@@ -95,8 +95,17 @@ def setup(self) -> None:
             self.downloader.download_checkpoint_if_necessary()
 
         self.encoder = self.get_encoder()
-        self.encoder.cuda()
-        self.encoder.eval()
+        if not self.is_finetune:
```

Comment:
May I propose to avoid replicating this logic in each child container? How about the following:
```python
class BaseMIL:
  def setup(self):
    # without the "if self.encoder_type ..." part
    self.encoder = self.get_encoder()
    ...

class DeepSMILECrck(BaseMIL):  # likewise for DeepSMILEPanda
  def setup(self):
    if self.encoder_type == SSLEncoder.__name__:
      ...
    super().setup()
```

Reply:
Thanks, this is a good idea. I have made this change now.